researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/factory/ArchiveFileFactory.java

```diff
@@ -69,7 +69,10 @@ public ArchiveFile fromResult(@NonNull Result result) {
         checkNotNull(result.getEndDate());
 
         if (result instanceof StepResult) {
-            return fromStepResult((StepResult) result);
+            // Skipped steps will have a null result
```

Comment:
Can we do this null handling in fromStepResult? 
We'd still need to check return value of fromStepResult to know whether to return early, but new code that calls fromStepResult without null checking first would no longer crash

Reply:
Sure, I'll move it there.